### PR TITLE
Post feature flag api

### LINKED ISSUE #18: [Backend] Create the POST of the feature Flag
Create the Route `POST` for posting the new feature flag created.

It should have the schema:

```
name : String
RoleUsers: String
createdBy: customerId
createdOn: timeStamp
isEnable: boolean
modifyBy: string
modifyOn: timestamp
```

### PR DESCRIPTION
**Pull request by Neha**
# Description

Created the POST and GET of the featureFlags. Used MongoDB instead of DynamobDB as of now.

Fixes #18 
#4 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] POSTMAN
- [X] Local browser

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/models/FeatureFlag.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const FeatureFlagSchema = schema({
+    name: {
+        type: String,
+        require: true
+    },
+    roleUsers: {
+        type: String,
+        require: true
+    },
```
Comment (pallabez) [inline backend/models/FeatureFlag.js]: It should be of type array of strings and I think just calling the field `roles` would suffice. What do you think?

File: backend/routes/featureFlag.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const FeatureFlagModel = require('../models/FeatureFlag.js');
+const router = express.Router();
+
+// Save the data in the DB
+router.post('/', (req,res) => {
+    const feature = new FeatureFlagModel(req.body);
+    feature.save(); 
+});
```
Comment (pallabez) [inline backend/routes/featureFlag.js]: Moving the function to a controller folder would be a good idea. And also returning 500 status code response if there is any unexpected errors.


```js
// controller/featureFlag.js
const addFeatureFlag = (req,res) => {
    try {
      const feature = new FeatureFlagModel(req.body);
      feature.save(); 
    } catch(err) {
      // returns 500 
    }
}

// routes/featureFlag.js
router.post('/',  addFeatureFlag);
```

File: backend/routes/index.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+const router = express.Router();
 
-/* GET home page. */
+/* GET FeatureFlags */
 router.get('/', function(req, res, next) {
-  res.render('index', { title: 'Express' });
+  FeatureFlagModel.find({}, function(err, flags) {
+    res.json(flags);
+ });
 });
```
Comment (pallabez) [inline backend/routes/index.js]: Would suggest the same here as above. Also, do we need `next`? If I'm not wrong you need it when you are making middle-wares.

Comment (Neha) [reply backend/routes/index.js]: As of now, I don't think so we need `next`. Let me think analyse a bit more and update the code (only if require)

File: backend/routes/featureFlag.js (hunk 2 above)
Comment (Neha) [reply backend/routes/featureFlag.js]: Are you suggesting moving just the logic or renaming the folder to `controller`

**Comment by Neha** [comment]
@pallabez  Please review. Comments are implemented

File: backend/routes/featureFlag.js (hunk 2 above)
Comment (pallabez) [reply backend/routes/featureFlag.js]: Moving the logic. 

Eg - https://github.com/Real-Dev-Squad/website-backend/blob/develop/routes/auctions.js

**Comment by Neha** [comment]
Done

**Review by pallabez — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**